### PR TITLE
Simplify Quarkus processor and fix the problem with unused beans

### PR DESCRIPTION
**Pull request by mkouba**

### CODE REVIEW

File: quarkus/user/src/main/resources/application.properties [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Make the ProcessorManager unremovable otherwise all Processor beans are marked as "unused" and get removed 
+quarkus.arc.unremovable-types=cz.ladicek.extdemo.framework.ProcessorManager
```
Comment (Ladicek) [inline quarkus/user/src/main/resources/application.properties:2]: Wouldn't it be better to do this using a build item in the extension?

Comment (Ladicek) [reply quarkus/user/src/main/resources/application.properties:2]: The `cdi-lite` variant likely has to do this too? And _there_, we'd have to use the config file.

Comment (mkouba) [reply quarkus/user/src/main/resources/application.properties:2]: > The `cdi-lite` variant likely has to do this too? And _there_, we'd have to use the config file.

I have no idea - I just tried to build the `quarkus/user` module and it failed.

Comment (mkouba) [reply quarkus/user/src/main/resources/application.properties:2]: > Wouldn't it be better to do this using a build item in the extension?

It depends. In a real app you would inject/use the `ProcessorManager` or ``Processor``s somewhere so...

Comment (Ladicek) [reply quarkus/user/src/main/resources/application.properties:2]: Funny, it works for me... :thinking:

Comment (Ladicek) [reply quarkus/user/src/main/resources/application.properties:2]: The test does inject `ProcessorManager` though?

Comment (mkouba) [reply quarkus/user/src/main/resources/application.properties:2]: > Funny, it works for me... thinking
> The test does inject ProcessorManager though?

The test itself passes but if you do `mvn clean install` and build the app then it fails...

**Pull request merged**